src/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -715,13 +720,33 @@ public with sharing class CRLP_RollupUI_SVC {
                         && !Test.isRunningTest()) {
                     throw new RollupException(Label.CRLP_DuplicateTargetField);
                 }
+
+                if (isRollupStateEnabled && rlp.DeveloperName == rollupCMTObj.recordName) {
+                    isRollupStale = rollupCMTObj.isDifferent(rlp);
+                    if (isRollupStale) {
+                        previousRollupType = new CRLP_RollupCMT.Rollup(rlp).getRollupType();
+                    }
+                    break;
```

Comment:
Why `break` here? Wouldn't that exit out of the loop before it finishes checking for duplicates?

Reply:
Good catch. The `break` should not be there. The suggested refactoring does not contain it.